crates/types/protobuf/restate/journal_events.proto

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2024 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+syntax = "proto3";
+
+package restate.journal.events;
+
+message TransientErrorEvent {
```

Comment:
We could use `Bilrost` to define the `TransientErrorEvent` type which would allow efficient serialization as well. It would avoid the extra moving from the domain type to the protobuf type.